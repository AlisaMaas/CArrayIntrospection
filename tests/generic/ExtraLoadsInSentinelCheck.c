/**
*
* Make sure we don't consider things to be null
* terminated if elements in the array are null terminated,
* but the array itself is not guaranteed null terminated.
*
**/

void find(char **arg) {
    find(arg + 7);
    for (int i = 0; *arg[i]; i++){
        find(arg);
    }
}