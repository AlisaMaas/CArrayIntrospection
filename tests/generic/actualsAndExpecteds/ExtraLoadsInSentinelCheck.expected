Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each function with the null-terminated annotation':
